### ISSUE TITLE
Logout from jupyterhub does not work properly with GithubOauth

### ISSUE THREAD

**Issue by leomora** [jupyterhub/oauthenticator#101]
Hello,

Expected behaviour:
When the user clicks the logout button, the /hub/login page should be displayed.

Issue:
Jupyterhub can't logout unless it logouts from github prevoiously. If the use is in the home page (/user/xxx/tree) and the logout button is pushed, jupyterhub is redirected to /hub and, after some redirections, it returns to the same page /user/xxx/tree

Jupyterhub version: 0.7.1
Oauth Authenticator vesion tested: 0.5.1/0.6.0
```
from oauthenticator.github import LocalGitHubOAuthenticator
c.JupyterHub.authenticator_class = LocalGitHubOAuthenticator
c.LocalGitHubOAuthenticator.oauth_callback_url = ''
c.LocalGitHubOAuthenticator.client_id = ''
c.LocalGitHubOAuthenticator.client_secret = ''
c.NotebookApp.open_browser = False
c.NotebookApp.ip='*'
c.NotebookApp.port = 8888
c.NotebookApp.token = u''
c.Authenticator.admin_users = {'hadoop'}
c.LocalAuthenticator.create_system_users = True
```
```
[I 2017-08-09 17:41:30.065 JupyterHub login:19] User logged out: xxxx
[I 2017-08-09 17:41:30.077 JupyterHub log:100] 302 GET /hub/logout (xxxx@yyy.yyyy.yyy.yyy) 12.40ms
[I 2017-08-09 17:41:30.437 JupyterHub log:100] 302 GET /hub/ (@yy.yyy.yyy.yyy) 1.75ms
[I 2017-08-09 17:41:30.680 JupyterHub log:100] 302 GET /oauth_login (@yy.yyy.yyy.yyy) 1.08ms
````
Thanks in advance,
Leandro

**Comment by minrk** [comment]
I believe this is fixed in JupyterHub master (will be 0.8)

**Comment by yuvipanda** [comment]
This is fixed now I think!
